src/rpc/client.cpp

```diff
@@ -314,56 +362,84 @@ static const CRPCConvertParam vRPCConvertParams[] =
-    std::set<std::pair<std::string, std::string>> membersByName;
-
 public:
-    CRPCConvertTable();
+    RPCConvertTable() = default;
 
-    /** Return arg_value as UniValue, and first parse it if it is a non-string parameter */
-    UniValue ArgToUniValue(std::string_view arg_value, const std::string& method, int param_idx)
+    const CRPCConvertParam* FromPosition(std::string_view method, int pos) const
     {
-        return members.count({method, param_idx}) > 0 ? Parse(arg_value) : arg_value;
+        auto it = std::ranges::find_if(vRPCConvertParams, [&](const auto& p) {
+            return p.methodName == method && p.paramIdx == pos;
+        });
+
+        return it == std::end(vRPCConvertParams) ? nullptr : &*it;
     }
 
-    /** Return arg_value as UniValue, and first parse it if it is a non-string parameter */
-    UniValue ArgToUniValue(std::string_view arg_value, const std::string& method, const std::string& param_name)
+    const CRPCConvertParam* FromName(std::string_view method, std::string_view name) const
     {
-        return membersByName.count({method, param_name}) > 0 ? Parse(arg_value) : arg_value;
+        auto it = std::ranges::find_if(vRPCConvertParams, [&](const auto& p) {
+            return p.methodName == method && p.paramName == name;
+        });
+
+        return it == std::end(vRPCConvertParams) ? nullptr : &*it;
     }
+
 };
 
-CRPCConvertTable::CRPCConvertTable()
+static const RPCConvertTable g_rpc_convert_table;
+
+static const CRPCConvertParam* ParamFromPosition(std::string_view method_name, size_t position)
 {
-    for (const auto& cp : vRPCConvertParams) {
-        members.emplace(cp.methodName, cp.paramIdx);
-        membersByName.emplace(cp.methodName, cp.paramName);
-    }
+    return g_rpc_convert_table.FromPosition(method_name, static_cast<int>(position));
+}
+
+static const CRPCConvertParam* ParamFromName(std::string_view method_name, std::string_view param_name)
+{
+    return g_rpc_convert_table.FromName(method_name, param_name);
+}
+
+/** Parse string to UniValue or throw runtime_error if string contains invalid JSON */
+static UniValue Parse(std::string_view raw)
+{
+    UniValue parsed;
+    if (!parsed.read(raw)) throw std::runtime_error(tfm::format("Error parsing JSON: %s", raw));
+    return parsed;
+}
+
+static UniValue ParseParam(const CRPCConvertParam* param, std::string_view raw)
+{
+    // Only parse parameters which have the JSON format; otherwise, treat them as strings.
+    return (param && param->format == ParamFormat::JSON) ? Parse(raw) : UniValue(std::string(raw));
 }
 
-static CRPCConvertTable rpcCvtTable;
 
 UniValue RPCConvertValues(const std::string &strMethod, const std::vector<std::string> &strParams)
 {
     UniValue params(UniValue::VARR);
 
-    for (unsigned int idx = 0; idx < strParams.size(); idx++) {
-        std::string_view value{strParams[idx]};
-        params.push_back(rpcCvtTable.ArgToUniValue(value, strMethod, idx));
+    for (std::string_view s: strParams) {
```

Comment:
In 9aaf55397159bb0e839c19d3d4578dfa144a059b "rpc: Handle -named argument parsing where '=' character is used"

nit: space between `s` and `:`
```suggestion
    for (std::string_view s : strParams) {
```

Reply:
Fixed.